ShellCheck/Analytics.hs

```diff
@@ -3670,6 +3670,33 @@ checkNonportableSignals _ = checkUnqualifiedCommand "trap" (const f)
         return $ err id 2173
             "SIGKILL/SIGSTOP can not be trapped."
 
+prop_checkMkdir0 = verify checkMkdir "mkdir -p -m 0755 dir"
+prop_checkMkdir1 = verify checkMkdir "mkdir -pm 0755 dir"
+prop_checkMkdir2 = verify checkMkdir "mkdir -vpm 0755 dir"
+prop_checkMkdir3 = verify checkMkdir "mkdir -pm 0755 -v dir"
+prop_checkMkdir4 = verifyNot checkMkdir "mkdir -p dir"
+prop_checkMkdir5 = verifyNot checkMkdir "mkdir -m 0755 dir"
+prop_checkMkdir6 = verifyNot checkMkdir "mkdir dir"
+checkMkdir _ = checkCommand "mkdir" (const $ f (Nothing,Nothing)) where
```

Comment:
My thought was that it would work like checkTr, growing to contain multiple checks over time. I'll change it to be more specific though since that seems to be what you prefer.